en/advanced_config/tuning_the_ecl_ekf.md

```diff
@@ -461,3 +461,17 @@ Note the NED velocity innovations and NE position innovations 'flat-line' after
 Note that after 10 seconds without GPS data, the EKF reverts back to a static position mode using the last known position and the NE position innovations start to change again.
 
 ![GPS Data Loss - in SITL](../../assets/ecl/gps_data_loss_-_velocity_innovations.png)
+
+### Barometer ground effect compensation
+If the vehicle has the tendency to climb back into the air when getting close to the ground during the landing phase the reason is most likely barometer ground effect.
+Air pushed down by the propellers hits the ground and creates a high pressure zone below the drone which causes a lower reading of pressure altitude. As a result an unwanted climb is commanded.
+The figure below shows a typical situation where the ground effect is present. Note how at the beginning and at the end of the flight the barometer signal dips.
+
+![Baromter ground effect](../../assets/ecl/gnd_effect.png)
+
+If you encouter this situation you can enable ground effect compensation. From the plot above try to read the magnitude of the baromter dip during takeoff or landing.
+Then set the parameter [EKF2_GND_EFF_DZ](../advanced_config/parameter_reference.md#EKF2_GND_EFF_DZ) to that value and add a 20 percent margin.
+
+If a terrain estimate is available (e.g. the vehicle is equipped with a range finder) then you can additionally specify the distance to the ground below which ground effect compensation should be activated.
```

Comment:
```suggestion
If a terrain estimate is available (e.g. the vehicle is equipped with a range finder) then you can instead use the distance to the ground below which ground effect compensation should be activated.
```

Reply:
@EliaTarasov I think I don't understand. Even if you have  a range finder you still need to specify EKF2_GND_EFF_DZ

Reply:
@RomanBapst My assumption was that in case we have terrain estimate then we always use it for takeoff and landing, but that might not be the case. We may use rangefinder to activate ground effect compensation, but still use baro as primary height source. Yes you are right sorry:)